x/farming/keeper/proposal_handler.go

```diff
@@ -1,54 +1,176 @@
 package keeper
 
 import (
-	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 
 	"github.com/tendermint/farming/x/farming/types"
 )
 
-// HandlePublicPlanProposal is a handler for executing a fixed amount plan creation proposal.
-func HandlePublicPlanProposal(ctx sdk.Context, k Keeper, plansAny []*codectypes.Any) error {
-	plans, err := types.UnpackPlans(plansAny)
-	if err != nil {
-		return err
+// HandlePublicPlanProposal is a handler for executing a public plan creation proposal.
+func HandlePublicPlanProposal(ctx sdk.Context, k Keeper, proposal *types.PublicPlanProposal) error {
+	if err := proposal.ValidateBasic(); err != nil {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
 	}
 
-	for _, plan := range plans {
-		switch p := plan.(type) {
-		case *types.FixedAmountPlan:
+	if proposal.AddRequestProposals != nil {
+		if err := k.AddPublicPlanProposal(ctx, proposal.AddRequestProposals); err != nil {
+			return err
+		}
+	}
+
+	if proposal.UpdateRequestProposals != nil {
+		if err := k.UpdatePublicPlanProposal(ctx, proposal.UpdateRequestProposals); err != nil {
+			return err
+		}
+	}
+
+	if proposal.DeleteRequestProposals != nil {
+		if err := k.DeletePublicPlanProposal(ctx, proposal.DeleteRequestProposals); err != nil {
+			return err
+		}
+	}
+
```

Comment:
I think we need to perform additional validation of public plans based on the state where all addition, removal, and modification have been made, For example, if the ratio sum of multiple Radio Plans with the same FarmingPoolAddress exceeds 1 (100%)

Reply:
We can check once `ValidateOverflowEpochRatio` on here, without on `Add and UpdatePublicPlanProposal` if reverted when `HandlePublicPlanProposal` return error.

checking `ValidateOverflowEpochRatio` the all each added, updated plan can cause a lot of duplicate for loop